guidance/models/_azureai_studio.py

```diff
@@ -164,27 +142,47 @@ def _generator(self, prompt, temperature: float):
 
             result_score = response_score.json()
 
-            chunk = result_score["output"]
-            encoded_chunk = chunk.encode("utf8")
+            chunk_score = result_score["output"]
+            encoded_chunk = chunk_score.encode("utf8")
             self.metrics.engine_input_tokens += input_token_count
-            self.metrics.engine_output_tokens += len(self.tokenizer(chunk))
-
-        # Now back to OpenAIChatEngine, with slight modifications since
-        # this isn't a streaming API
-        if temperature == 0:
-            cached_results = []
+            self.metrics.engine_output_tokens += len(self.tokenizer(chunk_score))
 
         yield encoded_chunk
 
-        if temperature == 0:
-            cached_results.append(encoded_chunk)
+    def _generator_completion(self, prompt, temperature: float):
+        prompt_decoded = prompt.decode("utf8")
+        parameters = dict(temperature=temperature, return_full_text=True)
+        payload = dict(
+            input_data=dict(input_string=[prompt_decoded], parameters=parameters)
+        )
 
-        # Cache the results after the generator is exhausted
-        if temperature == 0:
-            self.cache[cache_key] = cached_results
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": ("Bearer " + self._api_key),
+            "azureml-model-deployment": self._deployment,
+        }
+        response_score = requests.post(
+            self._endpoint,
+            json=payload,
+            headers=headers,
```

Comment:
Is this defaulted to streaming?

Reply:
No, this is a straight call/response.

Reply:
Does AzureAI Studio not support streaming on completions? We should always stream the generators if it's possible for lowest latency on long generations

Reply:
Am investigating